src/index.ts

```diff
@@ -39,21 +33,17 @@ module.exports = function isEqual (obj1, obj2, ignore, debug) {
     return true;
   }
   return false;
-};
+}
 
 /**
  * Determine if 'val1' and 'val2' are equal
- * @param {Object} val1
- * @param {Object} val2
- * @returns {Boolean}
  */
-function equal (val1, val2) {
-  const type1 = typeof val1;
-  const type2 = typeof val2;
+function equal(val1: any, val2: any): boolean {
 
-  // Convert NaN to null
-  if (type1 == 'number' && isNaN(val1)) val1 = null;
-  if (type2 == 'number' && isNaN(val2)) val2 = null;
+  // Special handling for NaN
+  if (typeof val1 == 'number' && isNaN(val1) && typeof val2 == 'number' && isNaN(val2)) {
```

Comment:
I think merging this into one test changes the logic a bit when e.g. `val1` is `NaN` and `val2` is `null`, which is why some testcases changed from `true` to `false`.